tasks.py

```diff
@@ -0,0 +1,206 @@
+) -> Path:
+    '''
+    Returns build directory where `cmake` shall be called from. Creates it and
+    possibly removes its contents (and artifacts_dir contents) if `clean=True`
+    is passed.
+    '''
+    build_dir = build_subdirectory
+    if clean:
+        remove_directory(build_dir)
+    build_dir.mkdir(parents=True, exist_ok=not clean)
+    return build_dir
+
+
+def _get_cmake_command(
+        build_dir: Path,
+        cmake_generator: str,
+        cmake_arch: Optional[str]=None,
+        config: str='Release',
+        verbose=False,
+):
+    '''
+    :param build_dir: Directory from where cmake will be called.
+    '''
+    root_dir = Path(__file__).parent
+    relative_root_dir = Path(os.path.relpath(root_dir, build_dir))
+    relative_artifacts_dir = Path(os.path.relpath(build_dir))
+
+    return strip_and_join(f"""
+        cmake
+            -G "{cmake_generator}"
+            {f'-A "{cmake_arch}"' if cmake_arch is not None else ""}
+            -DCMAKE_BUILD_TYPE={config}
+            -DCMAKE_INSTALL_PREFIX="{relative_artifacts_dir.as_posix()}"
+            {f'-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' if verbose else ''}
+            "{str(relative_root_dir)}"
+    """)
+
+
+def _get_wrappers_command(wrappers_dir: Path) -> str:
+    conda_prefix = os.environ['CONDA_PREFIX']
+    if sys.platform.startswith('win'):
+        autodiff_env_path = f"{conda_prefix}\\Library\\bin"
+    else:
+        autodiff_env_path = f"{conda_prefix}/bin"
+    return strip_and_join(f"""
+        create-wrappers
+            -t conda
+            --bin-dir {autodiff_env_path}
+            --dest-dir {wrappers_dir}
+            --conda-env-dir {conda_prefix}
+    """)
+
+
+def _get_test_command():
+    if sys.platform.startswith('win'):
+        test_command = strip_and_join(f"""
+            {BUILD_DIR_DEFAULT}\\test\\tests
+                --success
+                --reporter compact
+        """)
+    else:
+        test_command = strip_and_join(f"""
+            {BUILD_DIR_DEFAULT}/test/tests
+                --success
+                --reporter compact
+        """)
+    return test_command
+
+
+if sys.platform.startswith('win'):
+    @task
+    def msvc(c, clean=False, config='Release'):
+        """
+        Generates a Visual Studio project at the "build/msvc" directory.
+        Assumes that the environment is already configured using:
+            conda devenv
+            activate autodiff
+        """
+        build_dir, artifacts_dir = _get_and_prepare_build(
+            c,
+            clean=clean,
+            build_subdirectory=BUILD_DIR_DEFAULT / "msvc",
+        )
+        cmake_command = _get_cmake_command(build_dir=build_dir, cmake_generator="Visual Studio 15 2017",
+                                           cmake_arch="x64", config=config)
+        os.chdir(build_dir)
+        c.run(cmake_command)
+
+
+@task
+def compile(c, clean=False, config='Release', number_of_jobs=-1, verbose=False, gen_wrappers=False):
```

Comment:
This is the compile task. It builds and installs `autodiff`. All targets are considered.

Usage:
```bash
inv compile (options)
```

You can set some build configs with the following options:

* `-c` or `--clean`: set if a build is clean or not. A clean build removes the build dir before perform the build process. Default is `False`
* `--config`. Set if the build is Release of Debug mode. Default is `Release`
* `n` or `--number-of-jobs`. Number of processes to build and install `autodiff`. Default is `-1`, which is the serial option
* `-v` or `--verbose`. Set if the build and install task is verbose. Default is `False`
* `-g` or `--gen-wrappers`. Set if `conda` binaries should be wrapped (it means: symbolic links) to a dir inside build directory. I will give details about this task in another comment. Default is `False`